datafusion/common/Cargo.toml

```diff
@@ -34,19 +34,34 @@ path = "src/lib.rs"
 
 [features]
 avro = ["apache-avro"]
-default = []
+compression = ["xz2", "bzip2", "flate2", "zstd", "async-compression"]
+default = ["compression"]
 pyarrow = ["pyo3", "arrow/pyarrow"]
 
 [dependencies]
 apache-avro = { version = "0.15", default-features = false, features = ["snappy"], optional = true }
 arrow = { workspace = true }
 arrow-array = { workspace = true }
+async-compression = { version = "0.4.0", features = ["bzip2", "gzip", "xz", "zstd", "futures-io", "tokio"], optional = true }
+bytes = "1.4"
+bzip2 = { version = "0.4.3", optional = true }
 chrono = { version = "0.4", default-features = false }
+flate2 = { version = "1.0.24", optional = true }
+futures = "0.3"
 num_cpus = "1.13.0"
 object_store = { version = "0.6.1", default-features = false, optional = true }
 parquet = { workspace = true, optional = true }
 pyo3 = { version = "0.19.0", optional = true }
 sqlparser = { workspace = true }
+tokio = { version = "1.28", features = ["macros", "rt", "rt-multi-thread", "sync", "fs", "parking_lot"] }
```

Comment:
The only thing I am slightly worried about is adding dependencies to `datafusion_common`, but given how tokio is basically used all over the place in DataFusion this doesn't seem like it would result in a practical difference to to me

Reply:
Could you elaborate on the concern with adding dependencies to datafusion_common? I am working on #7322 currently which might add additional dependencies. I am defaulting to putting functionality in datafusion_common if it is useful in multiple other crates, but it could be potentially split out if this is a concern.

Reply:
I think some users of datafusion only use parts of it (like dask-sql only uses the parser / optimizer as I understand it). So I just worry that adding stuff to datafusion_common makes the dependency footprint for those projects larger

But I actually haven't checked the footprint / needs of other systems so I have no specific concerns or requirements I can share